test/helpers/proofs.js

```diff
@@ -0,0 +1,231 @@
+import Trie from 'merkle-patricia-tree'
+import utils from 'ethereumjs-util'
+import EthereumTx from 'ethereumjs-tx'
+import EthereumBlock from 'ethereumjs-block/from-rpc'
+
+const rlp = utils.rlp
+
+// raw header
+function getRawHeader(_block) {
+  if (typeof _block.difficulty !== 'string') {
+    _block.difficulty = '0x' + _block.difficulty.toString(16)
+  }
+
+  const block = new EthereumBlock(_block)
+  return block.header
+}
+
+// squanch transaction
+export function squanchTx(tx) {
+  tx.gasPrice = '0x' + parseInt(tx.gasPrice).toString(16)
+  tx.value = '0x' + parseInt(tx.value).toString(16) || '0'
+  tx.gas = '0x' + parseInt(tx.gas).toString(16)
+  tx.data = tx.input
+  return tx
+}
+
+function nibblesToTraverse(encodedPartialPath, path, pathPtr) {
+  let partialPath
+  if (
+    String(encodedPartialPath[0]) === '0' ||
+    String(encodedPartialPath[0]) === '2'
+  ) {
+    partialPath = encodedPartialPath.slice(2)
+  } else {
+    partialPath = encodedPartialPath.slice(1)
+  }
+
+  if (partialPath === path.slice(pathPtr, pathPtr + partialPath.length)) {
+    return partialPath.length
+  } else {
+    throw new Error('path was wrong')
+  }
+}
+
+export function getTxBytes(tx) {
+  const txObj = new EthereumTx(squanchTx(tx))
+  return txObj.serialize()
+}
+
+// build
+export async function getTxProof(tx, block, _web3 = web3) {
+  const txTrie = new Trie()
+  for (let i = 0; i < block.transactions.length; i++) {
+    const siblingTx = await _web3.eth.getTransaction(block.transactions[i])
```

Comment:
[2/2] it doesn't make any sense to do lots of web3/network calls here.

Reply:
Cleaned this up! Real nice catch. Test time increased though haha